### PR TITLE
Bootcamp at Algodaily

### PR DESCRIPTION
**Pull request by Shernz**
The day 0 problem of the Bootcamp - reverse a string.

### CODE REVIEW

File: src/com/leetcode/problem/easy/LengthOfLastWord.java [hunk 1]
```diff
@@ -33,10 +33,11 @@
 public class LengthOfLastWord {
 
     public static void main(String[] args) {
-        String[] s = {"Hello World", "    ", "awesome"};
+        String[] s = {"Hello World", "    ", "awesome", "   fly me   to   the moon  "};
         assert lengthOfLastWord(s[0]) == 5;
```
Comment (Xnkr) [inline src/com/leetcode/problem/easy/LengthOfLastWord.java:37]: Asserts belong inside tests. Rebase and fix.

File: src/main/java/com/algodaily/technicalInterviewPrep/ReverseAString.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+ *
+ * reverseString("reverseastring") should return "gnirtsaesrever".
+ *
+ * Constraints
+ * Do not use the built-in #reverse() method or [::-1] if Python
+ * Ideal solution would run in O(n) time complexity and O(1) space complexity
+ */
+public class ReverseAString {
+
+    public static String reverseString(String str) {
```
Comment (Xnkr) [inline src/main/java/com/algodaily/technicalInterviewPrep/ReverseAString.java:18]: Tidbit: Explore the source of StringBuilder#reverse and other ways of reversing a string - https://explainjava.com/reverse-string-java/

**Review by Xnkr — CHANGES_REQUESTED** [review]

File: src/main/java/com/algodaily/technicalInterviewPrep/ReverseAString.java (hunk 2 above)
Comment (Xnkr) [reply src/main/java/com/algodaily/technicalInterviewPrep/ReverseAString.java:18]: You'll notice something called "[Surrogate pairs](https://stackoverflow.com/a/54922477)"

Comment (Shernz) [reply src/main/java/com/algodaily/technicalInterviewPrep/ReverseAString.java:18]: In algodaily's solution, they had used recursion

Comment (Xnkr) [reply src/main/java/com/algodaily/technicalInterviewPrep/ReverseAString.java:18]: O(1) space complexity

**Review by Xnkr — APPROVED** [review]

**Pull request merged**